src/pharmpy/modeling/summary_individuals.py

```diff
@@ -0,0 +1,108 @@
+from typing import Union, Callable
+
+import pandas as pd
+import numpy as np
+
+from pharmpy.tools.modelsearch.tool import ModelSearchResults
+from pharmpy.model import Model
+
+from pharmpy.modeling.ml import predict_outliers
+from pharmpy.modeling.ml import predict_influential_individuals
+
+def summary_individuals(results: ModelSearchResults) -> Union[pd.DataFrame,None]:
+    """Creates a summary dataframe keyed by model-individual pairs for an input
+    ModelSearchResults object.
+
+    Parameters
+    ----------
+    results : ModelSearchResults
+        Input results object
+
+    Return
+    ------
+    pd.DataFrame | None
+        The summary as a dataframe
+
+    Examples
+    --------
+    >>> from pharmpy.modeling import *
+    >>> model = load_example_model("pheno")
+    >>> fit(model)
+    >>> results = run_tool('modelsearch', model=model, mfl='ABSORPTION(ZO);PERIPHERALS([1, 2])', algorithm='reduced_stepwise')
+    >>> summary_individuals(results) # doctest: +SKIP
+
+    """
+
+    if results.start_model is None: return None
+
+    models = [results.start_model] if results.models is None else [results.start_model, *results.models]
+
+    df = pd.concat(map(
+        groupedByIDAddColumnsOneModel,
+        models,
+    ), keys=[model.name for model in models], names=['model'], axis=0, verify_integrity=True)
+
+    df['diOFV'] = df.apply(lambda row: row['iOFV'] - df.loc[(row['parent_model'], row.name[1])]['iOFV'], axis=1)
+
+    return df
+
+def constant_series(model: Model, name: str, value) -> pd.Series:
+    index = model.dataset['ID'].unique()
+    series = pd.Series(np.repeat(value, len(index)), index=index, name=name)
+    series.index.rename('ID', inplace=True)
+    return series
```

Comment:
Maybe there is a better way?

Reply:
Yes, you can use broadcasting of one single value into a column: `df['mycolumn'] = np.nan`

Reply:
There is also `pd.DataFrame({'a': [1,2,3], 'b': np.nan})` instead of `pd.concat`.